hasher-matcher-actioner/webapp/src/settings/ActionRuleSettingsTab.jsx

```diff
@@ -138,13 +189,36 @@ export default function ActionRuleSettingsTab() {
                           action_id: actionRef.current.value,
                         };
 
-                        actionRules.push(newActionRule);
+                        setShowNameRequired(false);
+                        setShowNameMustBeUnique(false);
+                        setShowMustHaveLabelsRequired(false);
+                        setShowActionRequired(false);
 
-                        actionRules.sort((a, b) =>
-                          a.name.toLowerCase() > b.name.toLowerCase() ? 1 : -1,
-                        );
+                        if (actionRuleIsValid(newActionRule, actionRules)) {
+                          actionRules.push(newActionRule);
 
-                        setAdding(false);
+                          actionRules.sort((a, b) =>
+                            a.name.toLowerCase() > b.name.toLowerCase()
+                              ? 1
+                              : -1,
+                          );
+
+                          setAdding(false);
+                        } else {
+                          setShowNameRequired(!newActionRule.name);
+                          setShowNameMustBeUnique(
+                            !actionRuleNameIsUnique(
+                              nameRef.current.value,
+                              actionRules,
+                            ),
+                          );
+                          setShowMustHaveLabelsRequired(
+                            !newActionRule.must_have_labels,
+                          );
+                          setShowActionRequired(
+                            newActionRule.action_id === '0',
+                          );
+                        }
```

Comment:
nit: The onClick handler is getting too large to read easily. Consider moving it to its own function and simply calling that function in onClick?